File: apps/s-2-admin/src/api/workflow/Workflow.ts
import { User } from "../user/User";
import { WorkflowStep } from "../workflowStep/WorkflowStep";
import { WorkflowsOnEventType } from "../workflowsOnEventType/WorkflowsOnEventType";

export type Workflow = {
  id: number;
  name: string;
  user?: User;
  trigger?: "BEFORE_EVENT" | "EVENT_CANCELLED" | "NEW_EVENT";
  time: number | null;
  timeUnit?: "DAY" | "HOUR" | "MINUTE" | null;
  steps?: Array<WorkflowStep>;
  activeOn?: Array<WorkflowsOnEventType>;
};

File: apps/cactus-demo-dotnet/src/APIs/Workflow/Dtos/Workflow.cs
using CactusDemoDotnet.Core.Enums;

namespace CactusDemoDotnet.APIs.Dtos;

public class Workflow
{
    public int Id { get; set; }

    public string Name { get; set; }

    public int User { get; set; }

    public TriggerEnum Trigger { get; set; }

    public int? Time { get; set; }

    public TimeUnitEnum? TimeUnit { get; set; }

    public List<int>? Steps { get; set; }

    public List<int>? ActiveOn { get; set; }
}
